src/builder/actions/Actions.sol

```diff
@@ -676,6 +687,100 @@ library Actions {
         return (quarkOperation, action);
     }
 
+    function bridgeAcross(BridgeAsset memory bridge, PaymentInfo.Payment memory payment, bool useQuotecall)
+        internal
+        view
+        returns (IQuarkWallet.QuarkOperation memory, Action memory)
+    {
+        Accounts.ChainAccounts memory srcChainAccounts =
+            Accounts.findChainAccounts(bridge.srcChainId, bridge.chainAccountsList);
+
+        Accounts.ChainAccounts memory dstChainAccounts =
+            Accounts.findChainAccounts(bridge.destinationChainId, bridge.chainAccountsList);
+
+        Accounts.AssetPositions memory srcAssetPositions =
+            Accounts.findAssetPositions(bridge.assetSymbol, srcChainAccounts.assetPositionsList);
+
+        Accounts.AssetPositions memory dstAssetPositions =
+            Accounts.findAssetPositions(bridge.assetSymbol, dstChainAccounts.assetPositionsList);
+
+        Accounts.QuarkSecret memory accountSecret =
+            Accounts.findQuarkSecret(bridge.sender, srcChainAccounts.quarkSecrets);
+
+        bytes[] memory scriptSources = new bytes[](1);
+        scriptSources[0] = Across.bridgeScriptSource();
+
+        // Make FFI call to fetch a quote from Across API
+        bytes memory ffiCalldata = abi.encodeCall(
+            IFFI.requestAcrossQuote,
+            (
+                srcAssetPositions.asset,
+                dstAssetPositions.asset,
+                bridge.srcChainId,
+                bridge.destinationChainId,
+                bridge.amount
+            )
+        );
+        (bool success, bytes memory result) = ACROSS_FFI_ADDRESS.staticcall(ffiCalldata);
```

Comment:
Do we need staticcall here? I wonder if we mark the function as `pure` or `view` if we can get Solidity to do `STATICCALL` without resorting to low-level calls?

Reply:
E.g. https://docs.soliditylang.org/en/latest/contracts.html#view-functions

> View Functions[](https://docs.soliditylang.org/en/latest/contracts.html#view-functions)
> Functions can be declared view in which case they promise not to modify the state.
> **Note**
> If the compiler’s EVM target is Byzantium or newer (default) the opcode STATICCALL is used when view functions are called, which enforces the state to stay unmodified as part of the EVM execution. For library view functions DELEGATECALL is used, because there is no combined DELEGATECALL and STATICCALL. This means library view functions do not have run-time checks that prevent state modifications. This should not impact security negatively because library code is usually known at compile-time and the static checker performs compile-time checks.

Reply:
Oh interesting. I didn't know that the compiler uses STATICCALL under the hood for `pure/view` functions. Ok, I updated the code to no longer use `staticcall` directly so we can continue defining these functions as `pure`.